### PR TITLE
Do not use fork to invoke test scripts in pytorch rocm CI

### PR DESCRIPTION
**Pull request by bddppq**
#14497

### CODE REVIEW

**Comment by ezyang** [comment]
It shouldn't import common_utils either. run_test.py *literally* used to be a shell script, and it should stay that: a dumb shell script.

**Comment by bddppq** [comment]
Since this is going to be a temporary workaround (at the end amd should fix the underlying issue, otherwise distributed training will run into problem), I think it's not worth the effort of rewriting a 400 lines python script into a shell script. e.g. you will need to re-implement the TEST_WITH_ROCM and black_list logic (not saying it's hard, just it needs reimplementation).

File: test/common_methods_invocations.py [hunk 1]
```diff
@@ -97,6 +97,7 @@ def __len__(self):
 M = 10
 S = 5
 
+
```
Comment (ezyang) [inline test/common_methods_invocations.py]: Surprised this didn't cause lint to fail lol

File: test/test_distributions.py [hunk 2]
```diff
@@ -4398,7 +4399,7 @@ def f(sample, *values):
             values, sample = self._perturb(Dist, keys, values, sample)
             expected = f(sample, *values)
             actual = traced_f(sample, *values)
-            self.assertEqual(expected, actual,
+            self.assertEqual(expected, actual, allow_inf=True,
```
Comment (ezyang) [inline test/test_distributions.py]: What's up here?

**Review by ezyang — APPROVED** [review]
Looks ok, I guess? There's a feel sprinkled seeding calls which look a bit unmotivated. But I guess it's all fine since we're still doing regular old out-of-process for non-ROCm.

File: test/common_methods_invocations.py (hunk 1 above)
Comment (bddppq) [reply test/common_methods_invocations.py]: Because this wasn't a function before.

File: test/test_distributions.py (hunk 2 above)
Comment (bddppq) [reply test/test_distributions.py]: There is one elem is nan in both the expected and actual results (https://github.com/pytorch/pytorch/pull/14600#issuecomment-443105287).

**Pull request closed without merging**